### ISSUE TITLE
why fake rotate event not trigger OnRotate callback

### ISSUE THREAD

**Issue by cavanwang** [siddontang/go-mysql#448]
I implement & register the callback `OnRotate` and `OnRow`. I can get the log from:

```
 64     if ev.Header.LogPos == 0 {                                                  
 65       switch e := ev.Event.(type) {                                             
 66       case *replication.RotateEvent:                                            
 67         fakeRotateLogName = string(e.NextLogName)                               
 68         log.Infof("received fake rotate event, next log name is %s", e.NextLogName)
 69       }                                                                         
 70                                                                                 
 71       continue                                                                  
 72     } 
```

that indicates I received the rotate binlog-file-name event, but the framework not call `OnRotate` callback, it just `continue` to skip.

 Then my slave server received a few `OnRow` event callback, but I only know the binlog-position, not know current binlog-filename.

I think even if pos==0, we should also trigger `OnRotate`, framework can set pos to invalid value(e.g. -1) to indicate that this is a `binlog-filename-changed-notification`.

**Comment by gaojijun** [comment]
Yes, I think replication client does care about this fake rotate event. 

Just as the comment says:

 "Faked rotate event is required in a few cases, so slave can know which binlog the following events are from"
https://github.com/mysql/mysql-server/blob/8e797a5d6eb3a87f16498edcb7261a75897babae/sql/rpl_binlog_sender.h#L235

slave can know and need to know fake rotate event. So I don't think is a good idea skip this event.

In my case, at first, I don't know binlog file name, so my sync start from an empty name, with a faked rotate event, I can know the binlog file name and save it.

**Comment by lance6716** [comment]
Currently you can get the binlog name from `OnPosSynced` callbacks.
